--- ../src-base/minecraft/net/minecraft/pathfinding/NodeProcessor.java
+++ ../src-work/minecraft/net/minecraft/pathfinding/NodeProcessor.java
@@ -17,7 +17,7 @@
     protected boolean canOpenDoors;
     protected boolean canSwim;
 
-    public void init(IBlockAccess sourceIn, EntityLiving mob)
+    public synchronized void init(IBlockAccess sourceIn, EntityLiving mob)
     {
         this.blockaccess = sourceIn;
         this.entity = mob;
@@ -27,7 +27,7 @@
         this.entitySizeZ = MathHelper.floor(mob.width + 1.0F);
     }
 
-    public void postProcess()
+    public synchronized void postProcess()
     {
         this.blockaccess = null;
         this.entity = null;
@@ -47,7 +47,7 @@
         return pathpoint;
     }
 
-    public abstract PathPoint getStart();
+    public  abstract PathPoint getStart();
 
     public abstract PathPoint getPathPointToCoords(double x, double y, double z);
 
