--- ../src-base/minecraft/net/minecraft/block/BlockObserver.java
+++ ../src-work/minecraft/net/minecraft/block/BlockObserver.java
@@ -43,17 +43,19 @@
 
     public void updateTick(World worldIn, BlockPos pos, IBlockState state, Random rand)
     {
-        if (((Boolean)state.getValue(POWERED)).booleanValue())
-        {
-            worldIn.setBlockState(pos, state.withProperty(POWERED, Boolean.valueOf(false)), 2);
-        }
-        else
-        {
-            worldIn.setBlockState(pos, state.withProperty(POWERED, Boolean.valueOf(true)), 2);
-            worldIn.scheduleUpdate(pos, this, 2);
-        }
+        synchronized (this){
+            if (state.getValue(POWERED))
+            {
+                worldIn.setBlockState(pos, state.withProperty(POWERED, Boolean.FALSE), 2);
+            }
+            else
+            {
+                worldIn.setBlockState(pos, state.withProperty(POWERED, Boolean.TRUE), 2);
+                worldIn.scheduleUpdate(pos, this, 2);
+            }
 
-        this.updateNeighborsInFront(worldIn, pos, state);
+            this.updateNeighborsInFront(worldIn, pos, state);
+        }
     }
 
     public void neighborChanged(IBlockState state, World worldIn, BlockPos pos, Block blockIn, BlockPos fromPos)
@@ -145,4 +147,12 @@
     {
         return this.getDefaultState().withProperty(FACING, EnumFacing.getFront(meta & 7));
     }
+
+    /* ======================================== FORGE START =====================================*/
+    @Override
+    public void observedNeighborChange(IBlockState observerState, World world, BlockPos observerPos, Block changedBlock, BlockPos changedBlockPos)
+    {
+        observedNeighborChanged(observerState, world, observerPos, changedBlock, changedBlockPos);
+    }
+    /* ========================================= FORGE END ======================================*/
 }
